src/vmm/src/device_manager/persist.rs

```diff
@@ -226,14 +226,22 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                 }
                 TYPE_VSOCK => {
                     let vsock = locked_device
-                        .as_any()
+                        .as_mut_any()
                         // Currently, VsockUnixBackend is the only implementation of VsockBackend.
-                        .downcast_ref::<Vsock<VsockUnixBackend>>()
+                        .downcast_mut::<Vsock<VsockUnixBackend>>()
                         .unwrap();
+
                     let vsock_state = VsockState {
                         backend: vsock.backend().save(),
                         frontend: vsock.save(),
                     };
+
+                    // Send Transport event to reset connections if device
+                    // is activated.
+                    if vsock.is_activated() {
+                        vsock.send_transport_reset_event().unwrap();
```

Comment:
I would avoid unwrap() here and generally as a means of reporting errors to the user. We need to either report the error up the stack, write it in the log or exiting gracefully. It depends on how bad it is. 

Either move this call outside of the Persist trait impl or refactor a bit so it allows error reporting on save().